src/test/java/com/google/cloud/tools/test/utils/SpyVerifier.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.test.utils;
+
+import com.google.common.base.Preconditions;
+
+import org.mockito.Mockito;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Collections;
+import java.util.Map;
+
+public class SpyVerifier {
+  private final Object objectToInspect;
+  private final Class classToInspectAs;
+
+  public static SpyVerifier newVerifier(Object objectUnderInspection) {
```

Comment:
Just curious, any specific reason you couldn't just have gone for a simple constructor instead of the factory method?